budgie-fuzzyclock/fuzzyclock/FuzzyClockApplet.vala

```diff
@@ -0,0 +1,532 @@
+        cal_button.clicked.connect(on_cal_activate);
+
+        update_cal();
+
+        update_clock();
+        add(widget);
+        on_settings_change("clock-format");
+        popover.get_child().show_all();
+
+        show_all();
+    }
+
+    /**
+     * This is called to when the calendar app is updated/changed
+     */
+    void update_cal()
+    {
+        calprov = AppInfo.get_default_for_type(CALENDAR_MIME, false);
+        cal_button.set_sensitive(calprov != null);
+    }
+
+    /**
+     * This is called to launch Date and Time Settings app
+     */
+    void on_date_activate()
+    {
+        this.popover.hide();
+        var app_info = new DesktopAppInfo("gnome-datetime-panel.desktop");
+
+        if (app_info == null) {
+            return;
+        }
+        try {
+            app_info.launch(null, null);
+        } catch (Error e) {
+            message("Unable to launch gnome-datetime-panel.desktop: %s", e.message);
+        }
+    }
+
+    /**
+     * This is called to launch calendar app when user selects from drop-down
+     */
+    void on_cal_activate()
+    {
+        this.popover.hide();
+
+        if (calprov == null) {
+            return;
+        }
+        try {
+            calprov.launch(null, null);
+        } catch (Error e) {
+            message("Unable to launch %s: %s", calprov.get_name(), e.message);
+        }
+    }
+
+    public override void update_popovers(Budgie.PopoverManager? manager)
+    {
+        this.manager = manager;
+        manager.register_popover(widget, popover);
+    }
+
+    /**
+     * This is called when any of the preferences are changed
+     */
+    protected void on_settings_change(string key)
+    {
+        switch (key) {
+            case "clock-format":
+                SignalHandler.block((void*)this.clock_format, this.check_id);
+                ClockFormat f = (ClockFormat)settings.get_enum(key);
+                ampm = f == ClockFormat.TWELVE;
+                clock_format.set_active(f == ClockFormat.TWENTYFOUR);
+                this.update_clock();
+                SignalHandler.unblock((void*)this.clock_format, this.check_id);
+                break;
+            case "clock-show-date":
+                this.update_clock();
+                break;
+        }
+    }
+
+    /**
+     * Update the date if necessary
+     */
+    protected void update_date()
+    {
+        if (!check_date.get_active()) {
+            return;
+        }
+        string ftime;
+        // TRANSLATORS:  you may wish to visit this site for format string help
+        // https://valadoc.org/glib-2.0/GLib.DateTime.format.html
+        if (this.orient == Gtk.Orientation.HORIZONTAL) {
+            // TRANSLATORS: Horizontal Date prints "- <DayAbbr>, <MonAbbr> <DayNum>"
+            //              You may wish to localize for a more common formatting
+            //              Opt for spoken words over numerical values
+            ftime = _("- %a, %b %d");
```

Comment:
I feel like this is awkward to translate

Reply:
agreed. on the showtime applet we read the locale format and then calc the string to be displayed from the value returned. https://github.com/UbuntuBudgie/budgie-extras/blob/master/budgie-showtime/src/showtime_desktop/showtime_desktop.vala#L341

Reply:
I've pulled in the showtime date locale format calculation and used that instead of this translation